File: src/org/mtransit/android/analytics/AnalyticsEventsParamsProvider.java
package org.mtransit.android.analytics;

import androidx.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;

public final class AnalyticsEventsParamsProvider {

	@NonNull
	private final Map<String, Object> params;

	public AnalyticsEventsParamsProvider() {
		params = new HashMap<>();
	}

	@NonNull
	public AnalyticsEventsParamsProvider put(@NonNull String name, @NonNull String value) {
		params.put(name, value);
		return this;
	}

	@NonNull
	public AnalyticsEventsParamsProvider put(@NonNull String name, @NonNull Long value) {
		params.put(name, value);
		return this;
	}

	@NonNull
	public AnalyticsEventsParamsProvider put(@NonNull String name, @NonNull Double value) {
		params.put(name, value);
		return this;
	}

	@NonNull
	public Map<String, Object> to() {
		return params;
	}
}

File: app-android/src/main/java/org/mtransit/android/analytics/AnalyticsEventsParamsProvider.kt
package org.mtransit.android.analytics

class AnalyticsEventsParamsProvider {

    private val params = mutableMapOf<String, Any>()

    fun put(name: String, value: String): AnalyticsEventsParamsProvider {
        params[name] = value
        return this
    }

    fun put(name: String, value: Int) = put(name, value.toLong())

    fun put(name: String, value: Long): AnalyticsEventsParamsProvider {
        params[name] = value
        return this
    }

    fun put(name: String, value: Double): AnalyticsEventsParamsProvider {
        params[name] = value
        return this
    }

    fun to(): Map<String, Any> {
        return params
    }
}
